### PR TITLE
Add support for projections on symbolic names

### LINKED ISSUE #131: Add support for projections on symbolic names
To support queries like this in ([neo4j-graphql-java - Deeply nested query using temporal orderBy](https://github.com/neo4j-graphql/neo4j-graphql-java/blob/master/core/src/test/resources/movie-tests.adoc#deeply-nested-query-using-temporal-orderby)):
```cypher
MATCH (user:User)
RETURN user {
  knows:[sortedElement IN [(user)-[:KNOWS]->(userKnows:User) | userKnows {
    born: userKnows.born
  }] | sortedElement {
    born:{
      formatted: toString(sortedElement.born)
    }
  }]
}
```

the java code should look like:

```java
import org.neo4j.cypherdsl.core.*;
import org.neo4j.cypherdsl.core.renderer.Renderer;

class Scratch {
    public static void main(String[] args) {

        Node user = Cypher.node("User").named("user");
        Node userKnows = Cypher.node("User").named("userKnows");
        SymbolicName sortedElement = Cypher.name("sortedElement");

        PatternComprehension innerPatternComprehension = Cypher.listBasedOn(user.relationshipTo(userKnows, "KNOWS"))
                .returning(userKnows.project(
                        "born",
                        userKnows.property("born")
                ));
        StatementBuilder.BuildableStatement cypher = Cypher
                .match(user)
                .returning(
                        user.project(
                                "knows",
                                Cypher.listWith(sortedElement)
                                        .in(innerPatternComprehension)
                                        .returning(sortedElement.project(
                                                "born",
                                                Cypher.mapOf(
                                                        "formatted",
                                                        Cypher.call("toString").withArgs(Cypher.anyNode(sortedElement).property("born")).asFunction()
                                                )
                                        ))
                        ));
        System.out.println(Renderer.getDefaultRenderer().render(cypher.build()));
    }
}
```

### PR DESCRIPTION
**Pull request by Andy2003**
resolves #131

### CODE REVIEW

File: neo4j-cypher-dsl/src/main/java/org/neo4j/cypherdsl/core/SymbolicName.java [hunk 1]
```diff
@@ -40,7 +42,7 @@
  * @since 1.0
  */
 @API(status = Status.EXPERIMENTAL, since = "1.0")
-public class SymbolicName implements Expression {
+public class SymbolicName implements Expression, PropertyContainer {
```
Comment (michael-simons) [inline neo4j-cypher-dsl/src/main/java/org/neo4j/cypherdsl/core/SymbolicName.java:45]: That should not be necessary.

**Review by michael-simons — CHANGES_REQUESTED** [review]
I do like the change for the actual issue, I am not a fan of broadening the API of the `RETURN`, `WITH` and other methods in the same PR.

**Comment by Andy2003** [comment]
> I do like the change for the actual issue, I am not a fan of broadening the API of the `RETURN`, `WITH` and other methods in the same PR.

@michael-simons this was necessary otherwise there were conflicts b/c of the `PropertyContainer` which makes `SymbolicName` a `Named` and an `Expression`

So if I remove the `PropertyContainer` from `SymbolicName` I can revert all the other changes as well. 

Is that what you want?

**Comment by Andy2003** [comment]
> I do like the change for the actual issue, I am not a fan of broadening the API of the `RETURN`, `WITH` and other methods in the same PR.

@michael-simons this was necessary otherwise there were conflicts b/c of the `PropertyContainer` which makes `SymbolicName` a `Named` and an `Expression`

So if I remove the `PropertyContainer` from `SymbolicName` I can revert all the other changes as well. 

Is that what you want?

**Comment by Andy2003** [comment]
> I do like the change for the actual issue, I am not a fan of broadening the API of the `RETURN`, `WITH` and other methods in the same PR.

@michael-simons this was necessary otherwise there were conflicts b/c of the `PropertyContainer` which makes `SymbolicName` a `Named` and an `Expression`

So if I remove the `PropertyContainer` from `SymbolicName` I can revert all the other changes as well. 

Is that what you want?

**Comment by michael-simons** [comment]
Closed in favor of 804dc246dcd364ce217ec1a2bd6350653c22490d

**Pull request closed without merging**